### ISSUE TITLE
Loss not getting printed in when manual optimization is turned on

### ISSUE THREAD

**Issue by ParamsRaman** [Lightning-AI/pytorch-lightning#19476]
### Bug description

I ported an simple toy example in pytorch lightning and found that when manual optimization is turned on (to implement a custom optimizer), the loss doesn't get printed. Not sure if I am missing to set some something.

Any help is appreciated.

### What version are you seeing the problem on?

v2.1

### How to reproduce the bug
Baseline toy example (works fine and prints loss correctly):
```python

import pytorch_lightning as pl
import numpy as np
import torch
from torch.nn import MSELoss
from torch.optim import Adam
from ptadamw import AdamW
from torch.utils.data import DataLoader, Dataset
import torch.nn as nn


class SimpleDataset(Dataset):
    def __init__(self):
        X = np.arange(10000)
        y = X * 2
        X = [[_] for _ in X]
        y = [[_] for _ in y]
        self.X = torch.Tensor(X)
        self.y = torch.Tensor(y)

    def __len__(self):
        return len(self.y)

    def __getitem__(self, idx):
        return {"X": self.X[idx], "y": self.y[idx]}


class MyModel(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.fc = nn.Linear(1, 1)
        self.criterion = MSELoss()

    def forward(self, inputs_id, labels=None):
        outputs = self.fc(inputs_id)
        loss = 0
        if labels is not None:
            loss = self.criterion(outputs, labels)
        return loss, outputs

    def train_dataloader(self):
        dataset = SimpleDataset()
        return DataLoader(dataset, batch_size=1000)

    def training_step(self, batch, batch_idx):
        input_ids = batch["X"]
        labels = batch["y"]
        loss, outputs = self(input_ids, labels)
        return {"loss": loss}

    def configure_optimizers(self):
        optimizer = Adam(self.parameters())
        return optimizer


if __name__ == '__main__':
    model = MyModel()
    trainer = pl.Trainer(max_epochs=100, gpus=1)
    trainer.fit(model)

    X = torch.Tensor([[1.0], [51.0], [89.0]])
    _, y = model(X)
    print(y)

```

Ported toy example with manual optimization (runs without any errors but no loss being printed in the progress bar):
```
import pytorch_lightning as pl
import numpy as np
import torch
from torch.nn import MSELoss
from torch.optim import Adam
from sam import SAM
from ptadamw import AdamW
from torch.utils.data import DataLoader, Dataset
import torch.nn as nn


class SimpleDataset(Dataset):
    def __init__(self):
        X = np.arange(10000)
        y = X * 2
        X = [[_] for _ in X]
        y = [[_] for _ in y]
        self.X = torch.Tensor(X)
        self.y = torch.Tensor(y)

    def __len__(self):
        return len(self.y)

    def __getitem__(self, idx):
        return {"X": self.X[idx], "y": self.y[idx]}


class MyModel(pl.LightningModule):
    def __init__(self):
        super().__init__()
        self.fc = nn.Linear(1, 1)
        self.criterion = MSELoss()
        self.automatic_optimization = False

    def forward(self, inputs_id, labels=None):
        outputs = self.fc(inputs_id)
        loss = 0
        if labels is not None:
            loss = self.criterion(outputs, labels)
        return loss, outputs

    def train_dataloader(self):
        dataset = SimpleDataset()
        return DataLoader(dataset, batch_size=1000)

    def training_step(self, batch, batch_idx):
        optimizer = self.optimizers()

        # first forward-backward pass
        loss_1 = self.compute_loss(batch)
        self.manual_backward(loss_1)
        optimizer.first_step(zero_grad=True)

        # second forward-backward pass
        loss_2 = self.compute_loss(batch)
        self.manual_backward(loss_2)
        optimizer.second_step(zero_grad=True)

        return {"loss": loss_1}

    def compute_loss(self, batch):
        input_ids = batch["X"]
        labels = batch["y"]
        loss, outputs = self(input_ids, labels)
        return loss

    def configure_optimizers(self):
        base_optimizer = Adam
        optimizer = SAM(model.parameters(), base_optimizer, lr=0.01, rho=0.05)
        return optimizer


if __name__ == '__main__':
    model = MyModel()
    trainer = pl.Trainer(max_epochs=100, gpus=1)
    trainer.fit(model)

    X = torch.Tensor([[1.0], [51.0], [89.0]])
    _, y = model(X)
    print(y)

```
```


### Error messages and logs

Baseline code: time, it/s, loss and v_num fields are printed correctly
```
Epoch 99: 100%|█████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 203.64it/s, loss=4.16e+07, v_num=32]
`Trainer.fit` stopped: `max_epochs=100` reached.
Epoch 99: 100%|█████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 197.27it/s, loss=4.16e+07, v_num=32]
tensor([[ 1.9360],
        [46.3258],
        [80.0620]], grad_fn=<AddmmBackward0>)

```

Ported code: (loss is not being printed)
```
Epoch 99: 100%|████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 180.96it/s, v_num=33]
tensor([[  2.8524],
        [102.8746],
        [178.8915]], grad_fn=<AddmmBackward0>)
```


### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>


### More info

_No response_

**Comment by awaelchli** [comment]
Hey @ParamsRaman 
Thanks for reporting. At first this may look like a bug, and you may have been using an older version of Lightning. But as of version 2.0, we changed the progress bar to not automatically log the loss there. More details here: #16192

If you want to have it there, you will need to explicitly log it via `self.log("loss", loss, prog_bar=True)`. 
Closing this issue as it is the expected behavior (for both automatic and manual).
